flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -533,7 +600,25 @@ class CodeGenerator(
          |$reset
          |  }""".stripMargin
     }
-
+    
+    var existDistinct = false
+    if (distinctAggsFlags.isDefined){
+      val distAggsFlags = distinctAggsFlags.get
+      for(i <- distAggsFlags.indices){
+        if(distAggsFlags(i)){ existDistinct = true }
+      }
+    }
+    if(existDistinct){
+     val initReusMember = {
```

Comment:
The code for the distinct deduplication and tuple reusage for multi-argument aggregate function could go here.